mibidata/segmentation.py

```diff
@@ -49,35 +57,52 @@ def extract_cell_dataframe(label_image, image=None, mode='total'):
             if mode == 'total':
                 vals = image.data[nonzeros[0], nonzeros[1], :].sum(axis=0)
             elif mode == 'quadrant':
-                vals = _quadrant_mean(nonzeros, image)
+                vals = _circular_sectors_mean(nonzeros, image, num_sectors=4)
+            elif mode == 'circular_sectors':
+                vals = _circular_sectors_mean(nonzeros, image, num_sectors)
             else:
-                raise ValueError('"mode" must be either "total" or "quadrant"')
+                raise ValueError('"mode" must be either "total", "quadrant" or \
+                "circular_sectors"')
             row.extend(vals)
         rows.append(row)
     return pd.DataFrame(rows, columns=columns).set_index('label')
 
 
-def _quadrant_mean(inds, image):
-    """Gets the geometric mean of a regions's intensities across its quadrants.
+def _circular_sectors_mean(inds, image, num_sectors=8):
+    """Divide a region in circular sectors and get the geometric mean across the
+    sectors.
 
     Args:
         inds: A tuple of 2 arrays of the y- and x- indices of the pixels in a
             segmented region of an image.
         image: A MibiImage in which the corresponding pixel indices are located.
+        num_sectors: number of circular sectors to use. Optional, default is 8.
 
     Returns:
         An array whose length is equal to the number of channels in the image.
         Each value in the array is the geometric mean of the image's integrated
-        channel intensities over the regions's four quadrants.
+        channel intensities over the regions's num_sectors circular sectors.
     """
+    # calculate the geometric center of the cells and get the counts
     y_center, x_center = np.mean(inds, axis=1)
     vals = image.data[inds]  # has shape (num_pixels_in_cell, num_channels)
-    upper_left = vals[(inds[0] <= y_center) & (inds[1] < x_center)].sum(axis=0)
-    upper_right = vals[(inds[0] < y_center) & (inds[1] >= x_center)].sum(axis=0)
-    lower_left = vals[(inds[0] > y_center) & (inds[1] <= x_center)].sum(axis=0)
-    lower_right = vals[(inds[0] >= y_center) & (inds[1] > x_center)].sum(axis=0)
-    quads = np.stack((upper_left, upper_right, lower_left, lower_right), axis=1)
-    return np.power(np.product(quads, axis=1), 1 / 4)
+
+    # convert to polar coordinates: y, x -> phi, r
+    inds_pol = np.zeros_like(inds, dtype=float) # force float for accuracy
+    inds_pol[1], inds_pol[0] = util.car2pol(inds[1], inds[0],
+                                            x_center, y_center)
+
+    # create circular sectors
+    sectors = []
+    ang_step = 2.*np.pi/num_sectors
+    for i in range(num_sectors):
+        new_sector = vals[(inds_pol[0] >= i*ang_step)
+                          & (inds_pol[0] < (i + 1)*ang_step)].sum(axis=0)
+        sectors.append(new_sector)
+    secs = np.stack(sectors, axis=1)
```

Comment:
Here you are essentially binning a histogram with edges of angles, weighted by pixel counts, so would something like the following work?
```
edges = np.linspace(0, 2*np.pi, num_sectors+1)
secs = np.bincount(np.digitize(angles, edges),
                   weights=vals, minlength=num_sectors)
```

Reply:
Ok, I tried your method, with a few modifications. The major one being: a loop for each channel in the image. Unfortunately ```np.bincount``` only works in 1 dim. Please have a look at the new version and let me know your thoughts.

Reply:
@rfink: I reverted to the previous implementation because the check for empty sectors is more straightforward in the previous implementation. More details below in a new comment. Please let me know of your thoughts.

Reply:
It's fine to keep your implementation if it works better here due to the dimensionality etc. Thanks for looking into other options.